demo/lightning/checkpoint/simulated/llama2.py

```diff
@@ -0,0 +1,289 @@
+        dist.barrier()
+        print(f"Started iteration {i} for {operation} on rank {rank}...")
+        start_time = time.time()
+        if operation == 'save':
+            benchmark_strategy.save_checkpoint(distributed_state_dict,
+                                               filepath=checkpoint_path)
+        elif operation == 'load':
+            benchmark_strategy.load_checkpoint(
+                checkpoint_path=checkpoint_path,
+                initial_state_dict=template_state_dict)
+        dist.barrier()
+        end_time = time.time()
+        times.append(end_time - start_time)
+        print(f"Completed iteration {i} for {operation} on rank {rank}")
+    return times
+
+
+def run_benchmark(rank, world_size: int, project: str, filepath: str,
+                  sample_count: int, use_fsspec: bool,
+                  model_parameter_size: str) -> None:
+    setup(rank, world_size)
+
+    benchmark_strategy = BenchmarkStrategy(project=project,
+                                           path=filepath,
+                                           use_fsspec=use_fsspec)
+    print(f'Constructing state dict for LLAMA2 {model_parameter_size}')
+    state_dict = create_llama2_state_dict(world_size=world_size,
+                                          rank=rank,
+                                          parameters=model_parameter_size,
+                                          empty=False)
+
+    dist.barrier()
+    save_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'save',
+                                                      rank, world_size,
+                                                      model_parameter_size)
+
+    load_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'load',
+                                                      rank, world_size,
+                                                      model_parameter_size)
+
+    if rank == 0:
+        print(f"Time taken to save checkpoint:\
+                {statistics.mean(save_checkpoint_times):.4f} seconds (stdev {statistics.stdev(save_checkpoint_times):.4f})"
+              )
+        print(f"All save times: {save_checkpoint_times}")
+        print(f"Time taken to load checkpoint:\
+                 {statistics.mean(load_checkpoint_times):.4f} seconds (stdev {statistics.stdev(load_checkpoint_times):.4f})"
+              )
+        print(f"All load times: {load_checkpoint_times}")
+
+        tensor_size_per_instance = sum(v.element_size() * v.numel()
+                                       for v in state_dict.values())
+        total_size_bytes = tensor_size_per_instance * world_size
+
+        print(f"Size of distributed tensors (rank {rank}):\
```

Comment:
This probably shouldn't be behind `if rank == 0:`

Reply:
Same as above